src/coreclr/tools/Common/Compiler/NativeAotNameMangler.cs

```diff
@@ -52,8 +52,7 @@ public override string SanitizeName(string s, bool typeName = false)
 
                 if (((c >= 'a') && (c <= 'z')) || ((c >= 'A') && (c <= 'Z')))
```

Comment:
Does this target .NET 7? If so, you could also clean up such checks to use e.g. `char.IsAsciiLetter`.  Maybe we should consider a new analyzer for that.